<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="James Ware" />

<meta name="date" content="2016-08-01" />

<title>denovolyzeR intro</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">denovolyzeR intro</h1>
<h4 class="author"><em>James Ware</em></h4>
<h4 class="date"><em>2016-08-01</em></h4>



<div id="statistics-for-de-novo-variant-analysis" class="section level2">
<h2>Statistics for <em>de novo</em> variant analysis</h2>
<div id="introduction" class="section level4">
<h4>Introduction</h4>
<p>This package provides functions to analyse <em>de novo</em> genetic variants using the statistical framework described in <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4222185/">Samocha <em>et al</em> (2014) Nature Genetics 10.1038/ng.3050</a>. This vignette demonstrates the usage of the <code>denovolyzeR</code> package to recapitulate the analyses described in this paper:</p>
<ol style="list-style-type: decimal">
<li>What is the overall burden of <em>de novo</em> variation in the study cohort: are there more <em>de novos</em> than expected? As well as looking at the total burden of <em>de novos</em>, results are returned for different classes of variant, such as loss-of-function (lof), missense (mis) and synonymous (syn).</li>
<li>Do <em>de novo</em> variants cluster in specific genes: are there more genes containing multiple <em>de novos</em> than expected?</li>
<li>Are there any individual genes that contain more <em>de novos</em> than expected?</li>
</ol>
<p>If using the package, please cite <a href="http://www.ncbi.nlm.nih.gov/pubmed/26439716">Ware <em>et al</em> (2015) Curr Protoc Hum Genet. 10.1002/0471142905.hg0725s87</a>.</p>
</div>
</div>
<div id="installation" class="section level2">
<h2>Installation</h2>
<pre><code># Install the package if you haven't already.
# OPTION 1 - install the latest release from CRAN:
install.packages(&quot;denovolyzeR&quot;)  

# OPTION 2 - install the latest development version from GitHub.  Either download and install, or use devtools:
if(!&quot;devtools&quot; %in% installed.packages()){
  install.packages(&quot;devtools&quot;)
}
devtools::install_github(&quot;jamesware/denovolyzeR&quot;)</code></pre>
</div>
<div id="example-analysis" class="section level2">
<h2>Example analysis</h2>
<p>We start with a table of de novo variants. An example dataset is provided:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(denovolyzeR)
<span class="co"># have a look at the example data:</span>
<span class="kw">dim</span>(autismDeNovos)</code></pre></div>
<pre><code>## [1] 1040    2</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(autismDeNovos)</code></pre></div>
<pre><code>##      gene class
## 1  BCORL1   mis
## 2  SPANXD   mis
## 3   GLRA2   mis
## 4 RPS6KA3   non
## 5    TSR2   mis
## 6   GNL3L   syn</code></pre>
<div id="overall-de-novo-burden" class="section level4">
<h4>Overall <em>de novo</em> burden</h4>
<p>First we want to know whether there are more de novos than expected, using the <code>denovolyzeByClass()</code> function. These variants were obtained by sequencing 1,078 cases, so we use <code>nsamples=1078</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">denovolyzeByClass</span>(<span class="dt">genes=</span>autismDeNovos$gene,
                  <span class="dt">classes=</span>autismDeNovos$class,
                  <span class="dt">nsamples=</span><span class="dv">1078</span>)</code></pre></div>
<pre><code>##   class observed expected enrichment   pValue
## 1   syn      254    302.3      0.840 0.998000
## 2   mis      655    679.0      0.965 0.826000
## 3   lof      131     94.3      1.390 0.000199
## 4  prot      786    773.2      1.020 0.328000
## 5   all     1040   1075.5      0.967 0.864000</code></pre>
<p>The total number of de novos is almost exactly as our model predicts. However, we see a statistically significant excess of LOF variants in this population.</p>
</div>
<div id="genes-containing-multiple-de-novos" class="section level4">
<h4>Genes containing multiple <em>de novos</em></h4>
<p>Next, we look to see if the total number of genes that contain more than one de novo is greater than expected, using the <code>denovolyzeMultiHits()</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">denovolyzeMultiHits</span>(<span class="dt">genes=</span>autismDeNovos$gene,
                    <span class="dt">classes=</span>autismDeNovos$class,
                    <span class="dt">nsamples=</span><span class="dv">1078</span>)</code></pre></div>
<pre><code>##   class obs expMean expMax pValue nVars
## 1   syn   3     3.2      9   0.66   254
## 2   mis  31    20.5     30   0.00   655
## 3   lof   5     0.9      4   0.00   131
## 4  prot  43    27.7     43   0.01   786
## 5   all  66    47.9     65   0.00  1040</code></pre>
<p>obs = the number of genes in our dataset with &gt;1 <em>de novo</em> variant<br />
expMean = the expected number of genes containing &gt;1 <em>de novo</em>: an average obtained by permutation<br />
expMax = the maximum number of genes containing &gt;1 <em>de novo</em> in <code>nperms</code> permutations (default <code>nperms=100</code>)<br />
pValue = an empirical p value<br />
nVars = the total number of <em>de novo</em> variants in each class<br />
Note that the number of observed genes with &gt;1 protein-altering variant does <strong>not</strong> equal the number of genes with &gt;1 lof + number of genes with &gt;1 missense, as genes containing 1 lof + 1 missense will only be counted as “multihits” in the combined analysis.</p>
<p>Here it looks like there may be an excess of genes with &gt;1 lof variant, &gt;1 missense, and &gt;1 protein-altering variant. We will want to increase the number if permutations here to get a handle on our level of significance.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">denovolyzeMultiHits</span>(<span class="dt">genes=</span>autismDeNovos$gene,
                    <span class="dt">classes=</span>autismDeNovos$class,
                    <span class="dt">nsamples=</span><span class="dv">1078</span>,
                    <span class="dt">nperms=</span><span class="dv">1000</span>)</code></pre></div>
<pre><code>##   class obs expMean expMax pValue nVars
## 1   syn   3     3.3     10  0.646   254
## 2   mis  31    20.1     33  0.011   655
## 3   lof   5     1.1      6  0.006   131
## 4  prot  43    28.2     45  0.002   786
## 5   all  66    47.5     73  0.006  1040</code></pre>
<p>There is another important option here. The expected number of genes containing &gt;1 hit is obtained by permutation: Given n <em>de novo</em> variants, how many genes contain &gt;1 de novo? There are two options for selecting n: by default it is derived from your data: e.g. in the example above autismDeNovos contains 131 lof variants, so this is the number used in the permutation. This is controlled by the default parameter <code>nVars=&quot;actual&quot;</code></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sum</span>(autismDeNovos$class %in%<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;frameshift&quot;</span>,<span class="st">&quot;non&quot;</span>,<span class="st">&quot;splice&quot;</span>))</code></pre></div>
<pre><code>## [1] 131</code></pre>
<p>This is a conservative approach, addressing the question: “given the number of variants in our dataset, do we see more genes with &gt;1 variant than expected?”</p>
<p>An alternative approach simply asks whether there are more genes with &gt;1 variant than our de novo model predicts. This is accessed by setting <code>nVars=&quot;expected&quot;</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">denovolyzeMultiHits</span>(<span class="dt">genes=</span>autismDeNovos$gene,
                    <span class="dt">classes=</span>autismDeNovos$class,
                    <span class="dt">nsamples=</span><span class="dv">1078</span>,
                    <span class="dt">nperms=</span><span class="dv">1000</span>,
                    <span class="dt">nVars=</span><span class="st">&quot;expected&quot;</span>)</code></pre></div>
<pre><code>##   class obs expMean expMax pValue      nVars
## 1   syn   3     4.5     12  0.848  302.27646
## 2   mis  31    21.3     35  0.020  678.98816
## 3   lof   5     0.5      4  0.000   94.26139
## 4  prot  43    27.1     44  0.002  773.24955
## 5   all  66    50.6     73  0.011 1075.52601</code></pre>
</div>
<div id="do-any-individual-genes-contain-more-de-novos-than-expected" class="section level4">
<h4>Do any individual genes contain more <em>de novos</em> than expected</h4>
<p>We see 5 genes containing &gt;1 <em>de novo</em> lof variant. This is more than expected, but are any of these genes individually significant? We can <code>denovolyzeByGene()</code> to find out.</p>
<p>By default this function compares the number of LOF variants against expectation for each gene, and then the total number of protein-altering variants (LOF + missense). It can also be configured to return other classes if relevant.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(
<span class="kw">denovolyzeByGene</span>(<span class="dt">genes=</span>autismDeNovos$gene,
                 <span class="dt">classes=</span>autismDeNovos$class,
                 <span class="dt">nsamples=</span><span class="dv">1078</span>)
  )</code></pre></div>
<pre><code>##         gene lof_observed lof_expected lof_pValue prot_observed
## 239   DYRK1A            3            0   2.69e-08             3
## 706    SCN2A            3            0   1.83e-06             5
## 141     CHD8            3            0   7.19e-07             4
## 674     RFX8            0            0   1.00e+00             2
## 796 SUV420H1            1            0   6.37e-03             3
## 618     POGZ            2            0   1.23e-04             2
##     prot_expected prot_pValue
## 239           0.1    2.77e-05
## 706           0.1    3.70e-07
## 141           0.2    2.44e-05
## 674           0.0    2.34e-05
## 796           0.1    3.17e-05
## 618           0.1    5.07e-03</code></pre>
<p>Several genes meet statistical significance after correcting for multiple testing. Default options apply two tests across 19618 genes, so a Bonferroni corrected p-value threshold at <span class="math inline">\(\alpha\)</span> = 0.05 would be <span class="math inline">\(1.3\times 10^{-6}\)</span>.</p>
</div>
<div id="geneset-analysis" class="section level4">
<h4>Geneset analysis</h4>
<p>The analyses presented so far have been exome-wide. It may be appropriate to restrict analyses to a geneset of interest - for example, it may be relevant to examine the burden of <em>de novo</em> variation in a pathway of interest, or initial variant detection may have been restricted to a set of candidate genes (rather than whole exome sequencing). All of the above funtions can be targeted to a subset of genes using the <code>includeGenes</code> argument.</p>
<p>The package includes as an example a list of 837 genes that interact with the fragile X mental retardation protein (FMRP). Is this geneset enriched for <em>de novos</em>, and recurrent <em>de novos</em>, in our autism trios?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nrow</span>(fmrpGenes); <span class="kw">head</span>(fmrpGenes)</code></pre></div>
<pre><code>## [1] 837</code></pre>
<pre><code>##            ensgID          enstID hgncID hgncSymbol geneName
## 1 ENSG00000142599 ENST00000337907   9965       RERE     RERE
## 2 ENSG00000149527 ENST00000449969  29037      PLCH2    PLCH2
## 3 ENSG00000078369 ENST00000378609   4396       GNB1     GNB1
## 4 ENSG00000157933 ENST00000378536  10896        SKI      SKI
## 5 ENSG00000171735 ENST00000303635  18806     CAMTA1   CAMTA1
## 6 ENSG00000188157 ENST00000379370    329       AGRN     AGRN</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">denovolyzeByClass</span>(<span class="dt">genes=</span>autismDeNovos$gene,
                  <span class="dt">classes=</span>autismDeNovos$class,
                  <span class="dt">nsamples=</span><span class="dv">1078</span>,
                  <span class="dt">includeGenes=</span>fmrpGenes$geneName)</code></pre></div>
<pre><code>##   class observed expected enrichment   pValue
## 1   syn       28     33.6      0.835 8.53e-01
## 2   mis       83     74.4      1.110 1.74e-01
## 3   lof       32      9.1      3.500 3.18e-09
## 4  prot      115     83.6      1.380 6.47e-04
## 5   all      143    117.1      1.220 1.13e-02</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">denovolyzeMultiHits</span>(<span class="dt">genes=</span>autismDeNovos$gene,
                    <span class="dt">classes=</span>autismDeNovos$class,
                    <span class="dt">nsamples=</span><span class="dv">1078</span>,
                    <span class="dt">nperms=</span><span class="dv">1000</span>,
                    <span class="dt">includeGenes=</span>fmrpGenes$geneName)</code></pre></div>
<pre><code>## Warning in denovolyzeMultiHits(genes = autismDeNovos$gene, classes =
## autismDeNovos$class, : De novo list includes 897 genes not specified for
## inclusion. These will not be analysed.</code></pre>
<pre><code>##   class obs expMean expMax pValue nVars
## 1   syn   0     0.6      4  1.000    28
## 2   mis   9     5.1     11  0.061    83
## 3   lof   3     1.0      6  0.061    32
## 4  prot  15     9.4     18  0.028   115
## 5   all  20    14.3     25  0.045   143</code></pre>
</div>
</div>
<div id="other-functions" class="section level2">
<h2>Other functions</h2>
<p><code>viewProbabilityTable</code> provides access to the underlying <em>de novo</em> probability tables used to calculate expected <em>de novo</em> burdens throughout this package:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(
  <span class="kw">viewProbabilityTable</span>()
  )</code></pre></div>
<pre><code>##   hgncID hgncSymbol          enstID          ensgID geneName          syn
## 1    658       ARF5 ENST00000000233 ENSG00000004059     ARF5 2.704295e-06
## 2   6752       M6PR ENST00000000412 ENSG00000003056     M6PR 2.877349e-06
## 3   3720      FKBP4 ENST00000001008 ENSG00000004478    FKBP4 5.608786e-06
## 4  20581    CYP26B1 ENST00000001146 ENSG00000003137  CYP26B1 1.013847e-05
## 5  28816    NDUFAF7 ENST00000002125 ENSG00000003509  NDUFAF7 5.079037e-06
## 6   4008      FUCA2 ENST00000002165 ENSG00000001036    FUCA2 5.445799e-06
##            mis          non       splice   frameshift          lof
## 1 6.234164e-06 2.016875e-07 2.207269e-07 1.059040e-06 1.481454e-06
## 2 7.767629e-06 5.370605e-07 3.469603e-07 7.571830e-08 9.597391e-07
## 3 1.346427e-05 7.503486e-07 3.996519e-07 4.035024e-08 1.190351e-06
## 4 2.091603e-05 4.151766e-07 2.143277e-07 6.023546e-07 1.231859e-06
## 5 1.208114e-05 7.784025e-07 3.826662e-07 5.197949e-07 1.680864e-06
## 6 1.222216e-05 8.668046e-07 2.605275e-07 1.396694e-06 2.524026e-06
##           prot          all
## 1 7.715618e-06 1.041991e-05
## 2 8.727368e-06 1.160472e-05
## 3 1.465462e-05 2.026341e-05
## 4 2.214789e-05 3.228636e-05
## 5 1.376201e-05 1.884104e-05
## 6 1.474619e-05 2.019199e-05</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(
  <span class="kw">viewProbabilityTable</span>(<span class="dt">format=</span><span class="st">&quot;long&quot;</span>)
  )</code></pre></div>
<pre><code>##   hgncID hgncSymbol          enstID          ensgID geneName class
## 1    658       ARF5 ENST00000000233 ENSG00000004059     ARF5   syn
## 2   6752       M6PR ENST00000000412 ENSG00000003056     M6PR   syn
## 3   3720      FKBP4 ENST00000001008 ENSG00000004478    FKBP4   syn
## 4  20581    CYP26B1 ENST00000001146 ENSG00000003137  CYP26B1   syn
## 5  28816    NDUFAF7 ENST00000002125 ENSG00000003509  NDUFAF7   syn
## 6   4008      FUCA2 ENST00000002165 ENSG00000001036    FUCA2   syn
##          value
## 1 2.704295e-06
## 2 2.877349e-06
## 3 5.608786e-06
## 4 1.013847e-05
## 5 5.079037e-06
## 6 5.445799e-06</code></pre>
<p>Most of the core functionality of this package is contained in the <code>denovolyze</code> function. The <code>denovolyzeByClass</code> and <code>denovolyzeByGene</code> functions used in this vignette are convenience functions that set defaults appropriate to the most common usages of this function. Full details of additional options and default behaviours are available using <code>?denovolyze</code>.</p>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
